Release notes 8.11 (#11031)
coq/coq: doc/sphinx/changes.rst

  Amorim).

**Commands and options**

- **Removed.**
  Deprecated flag `Refine Instance Mode`
  (`#9530 <https://github.com/coq/coq/pull/9530>`_, fixes
  `#3632 <https://github.com/coq/coq/issues/3632>`_, `#3890
  <https://github.com/coq/coq/issues/3890>`_ and `#4638
  <https://github.com/coq/coq/issues/4638>`_
  by Maxime Dénès, review by Gaëtan Gilbert).
- **Removed.**
  Undocumented :n:`Instance : !@type` syntax
  (`#10185 <https://github.com/coq/coq/pull/10185>`_, by Gaëtan Gilbert).
- **Removed.**
  Deprecated ``Show Script`` command
  (`#10277 <https://github.com/coq/coq/pull/10277>`_, by Gaëtan Gilbert).
- **Added.**
  Unsafe commands to enable/disable guard checking, positivity checking
  and universes checking (providing a local `-type-in-type`).
  See :ref:`controlling-typing-flags`
  (`#10291 <https://github.com/coq/coq/pull/10291>`_ by Simon Boulier).
- **Fixed.**
  Two bugs in :cmd:`Export`. This can have an impact on the behavior of the
  :cmd:`Import` command on libraries. `Import A` when `A` imports `B` which exports
  `C` was importing `C`, whereas :cmd:`Import` is not transitive. Also, after
  `Import A B`, the import of `B` was sometimes incomplete.

>   (`#10476 <https://github.com/coq/coq/pull/10476>`_, by Maxime Dénès).

Author: Despite being a fix, this should probably be part of the highlights, and we could add a `.. warning::` to signal that this bug fix can break a lot of scripts.

   .. include:: ../unreleased.rst

Version 8.11
------------

Summary of changes
~~~~~~~~~~~~~~~~~~

The main changes brought by |Coq| version 8.11 are:

- `Ltac2`__, a new tactic language for writing more robust larger scale
  tactics, with built-in support for datatypes and the multi-goal tactic monad.
- `Primitive floats`__ are integrated in terms and follow the binary64 format
  of the IEEE 754 standard, as specified in the `Coq.Float.Floats` library.
- `Cleanups`__ of the section mechanism, delayed proofs and further
  restrictions of template polymorphism to fix soundness issues related to
  universes.
- New `unsafe flags`__ to disable locally guard, positivity and universe
  checking. Reliance on these flags is always printed by
  :g:`Print Assumptions`.

> - `Fixed bugs`__ of :g:`Export` and :g:`Import` that can have an impact

Author: I suggest making this warning even more visible:

```suggestion
- `Fixed bugs`__ of :g:`Export` and :g:`Import` that can have a **significant impact**
```

We could also add a `.. warning::` box to the detailed entry below.

  on user developments.
- New interactive development method based on `vos` `interface files`__,
  allowing to work on a file without recompiling the proof parts of their
  dependencies.
- New :g:`Arguments` annotation for `bidirectional type inference`__
  configuration for reference (e.g. constants, inductive) applications.
- Generalization of the :g:`under` and :g:`over` tactics__ of SSReflect to
  arbitrary relations.
- `Revision`__ of the :g:`Coq.Reals` library, its axiomatisation and
  instances of the constructive and classical real numbers.

__ 811Ltac2_
__ 811PrimitiveFloats_
__ 811Sections_
__ 811UnsafeFlags_
__ 811ExportBug_
__ 811vos_
__ 811BidirArguments_
__ 811SSRUnderOver_
__ 811Reals_

The ``dev/doc/critical-bugs`` file documents the known critical bugs of |Coq|
and affected releases. See the `Changes in 8.11+beta1`_ section for the
detailed list of changes, including potentially breaking changes marked with

> **Changed.**.

Author: The repetition of `.` is weird. Moreover, Gaëtan has suggested making that a `:`, which I've already implemented in #10931.

```suggestion
**Changed**.
```

Maxime Dénès, Emilio Jesús Gallego Arias, Gaëtan Gilbert, Michael
Soegtrop, Théo Zimmermann worked on maintaining and improving the
continuous integration system and package building infrastructure.

Coq's documentation is available at https://coq.github.io/doc/V8.11+beta1/api (documentation of

> the ML API), https://coq.github.io/doc/v8.11+beta1/refman (reference

Author: ```suggestion
the ML API), https://coq.github.io/doc/V8.11+beta1/refman (reference
```

> manual), and https://coq.github.io/doc/v8.11+beta1/stdlib (documentation of

Author: ```suggestion
manual), and https://coq.github.io/doc/V8.11+beta1/stdlib (documentation of
```

the standard library).

The OPAM repository for |Coq| packages has been maintained by
Karl Palmskog, Matthieu Sozeau, Enrico Tassi with contributions
from many users. A list of packages is available at
https://coq.inria.fr/opam/www/.

The ?? contributors to this version are

Many power users helped to improve the design of the new features via
the issue and pull request system, the |Coq| development mailing list,

> the coq-club@inria.fr mailing list or the Discourse forum. It would

Author: ```suggestion
the coq-club@inria.fr mailing list or the `Discourse forum <https://coq.discourse.group/>`_. It would
```

  closes `#8276 <https://github.com/coq/coq/issues/8276>`_,
  by Guillaume Bertholon, Erik Martin-Dorel, Pierre Roux).
- **Changed.**
  Internal definitions generated by :tacn:`abstract`\-like tactics are now inlined
  inside universe :cmd:`Qed`\-terminated polymorphic definitions, similarly to what
  happens for their monomorphic counterparts,
  (`#10439 <https://github.com/coq/coq/pull/10439>`_, by Pierre-Marie Pédrot).

  .. _811Sections:

- **Fixed.**
  Section data is now part of the kernel. Solves a soundness issue
  in interactive mode where global monomorphic universe constraints would be
  dropped when forcing a delayed opaque proof inside a polymorphic section. Also
  relaxes the nesting criterion for sections, as polymorphic sections can now
  appear inside a monomorphic one
  (`#10664, <https://github.com/coq/coq/pull/10664>`_ by Pierre-Marie Pédrot).
- **Changed.**
  Using ``SProp`` is now allowed by default, without needing to pass
  ``-allow-sprop`` or use :flag:`Allow StrictProp` (`#10811
  <https://github.com/coq/coq/pull/10811>`_, by Gaëtan Gilbert).

**Specification language, type inference**

  .. _811BidirArguments:

- **Added.**

> 

Author: Is this newline accidental?

  documentation. (`#10441 <https://github.com/coq/coq/pull/10441>`_,
  by Pierre-Marie Pédrot)
- **Added:**
  The :cmd:`Section` vernacular command now accepts the "universes" attribute. In
  addition to setting the section universe polymorphism, it also locally sets
  the universe polymorphic option inside the section.
  (`#10441 <https://github.com/coq/coq/pull/10441>`_, by Pierre-Marie Pédrot)
- **Fixed:**
  ``Program Fixpoint`` now uses ``ex`` and ``sig`` to make telescopes
  involving ``Prop`` types (`#10758
  <https://github.com/coq/coq/pull/10758>`_, by Gaëtan Gilbert, fixing
  `#10757 <https://github.com/coq/coq/issues/10757>`_ reported by
  Xavier Leroy).
- **Changed:**
  Output of the :cmd:`Print` and :cmd:`About` commands.
  Arguments meta-data is now displayed as the corresponding
  :cmd:`Arguments <Arguments (implicits)>` command instead of the
  human-targeted prose used in previous Coq versions. (`#10985
  <https://github.com/coq/coq/pull/10985>`_, by Gaëtan Gilbert).
- **Added:** ``#[refine]`` attribute for :cmd:`Instance`, a more
  predictable version of the old ``Refine Instance Mode`` which
  unconditionally opens a proof (`#10996

>   <https://github.com/coq/coq/pull/10996>`_, by Gaëtan Gilbert).

Author: Maybe worth highlighting in the summary.

- **Changed:**
  Moved the :tacn:`auto` hints of the `OrderedType` module into a new `ordered_type`
  database
  (`#9772 <https://github.com/coq/coq/pull/9772>`_,
  by Vincent Laporte).
- **Removed:**
  Deprecated modules `Coq.ZArith.Zlogarithm` and `Coq.ZArith.Zsqrt_compat`
  (`#9881 <https://github.com/coq/coq/pull/9811>`_,
  by Vincent Laporte).

  .. _811Reals:

- **Added:**
  Module `Reals.ConstructiveCauchyReals` defines constructive real numbers
  by Cauchy sequences of rational numbers
  (`#10445 <https://github.com/coq/coq/pull/10445>`_, by Vincent Semeria,
  with the help and review of Guillaume Melquiond and Bas Spitters).

- **Changed:** Classical real numbers are now defined
  as a quotient of these constructive real numbers, which significantly reduces
  the number of axioms needed (see `Reals.Rdefinitions` and `Reals.Raxioms`),
  while preserving backward compatibility.

  Futhermore, the new axioms for classical real numbers include the limited
  principle of omniscience (`sig_forall_dec`), which is a logical principle
  instead of an *ad hoc* property of the real numbers.

  (`#10445 <https://github.com/coq/coq/pull/10445>`_, by Vincent Semeria,

>   with the help and review of Guillaume Melquiond and Bas Spitters)

Author: This whole bullet-point is made obsolete by the next one, so this should probably simply be removed. The part "which significantly reduces the number of axioms needed (see `Reals.Rdefinitions` and `Reals.Raxioms`), while preserving backward compatibility" is the interesting part that could be copy-pasted to the next bullet point.
